File: src/main/java/pl/presz/expenseManager/model/Transaction.java
package pl.presz.expenseManager.model;


import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;


@Entity
@Getter
@Setter
@Table(name ="transactions")
public class Transaction {

    @Id
    private long transactionId;

    private LocalDateTime createDate;

    private LocalDate transactionDate;

    private BigDecimal amount;

    @OneToOne(fetch=FetchType.EAGER)
    @JoinColumn(name="paymentId")
    private Payment payment;

    private BigDecimal amountLeftWithoutExpens;




}

File: user-app/src/app/transaction/transaction.ts
export class Transaction {

  transactionId: number;
  amount: number;
  transactionDate: Date;
  paymentMethod: String;

  constructor(transactionId: number, amount: number, transactionDate: Date, paymentMethod: String ){
    this.transactionId = transactionId;
    this.amount = amount;
    this.transactionDate = transactionDate;
    this.paymentMethod = paymentMethod;
  }

}
